docs/porting/target/tickless/tickless.md

```diff
@@ -0,0 +1,64 @@
+
+- Sleep mode: Both low and high-frequency clocks are active.
+- Deep sleep mode: Only the low-frequency clock is active.
+
+To enable tickless mode in Mbed OS, your target has to meet following requirements:
+
+- Support for sleep HAL API
+- Support for low power or microsecond ticker HAL API
+
+If the target supports both low power and microsecond ticker, then low power ticker is selected with tickless mode.
+
+For instance, the system runs only with two threads thread 1 and thread 2. During the first 2 tick interrupts thread 1 and thread 2 schedules alternatively. When thread 1 entered into a wait state, the scheduler preempts and schedules thread 2 for execution. After some time, thread 2 enters into a wait state and get preempted. Now scheduler schedules idle thread to run as two threads are in the wait state. This idle thread put the system to sleep. After multiple SysTick periodic interrupts, thread 2 is rescheduled for execution. All previous Systick wake-ups waste power.
+
+![](./resources/Normal_Tick.png)
+
+But with tickless mode, it suspends the SysTick periodic wake up as no threads are ready to run for quite some time which inturn saves the power.
+
+![](./resources/Tickless.png)
+
+### Enable tickless mode support in full profile
+
+Find your target in `mbed-os/targets/targets.json` and update its configuration as described below.
+
+1. Add `MBED_TICKLESS` macro into the `macros` option as follows:
+
+    ```json
+	"macros": [
+        "MBED_TICKLESS"
+    ]    
+    ```
+1. If your target does not have a low power ticker or you prefer to use microsecond ticker instead of low power ticker for tickless mode, then ensure that tickless is enabled using the microsecond ticker:
+
+    ```json
+    "overrides": {
+        "tickless-from-us-ticker": true
+    }
+    ```
+
+Targets supporting tickless mode override the default SysTick mechanism and use the [OsTimer](../mbed-os-api-doxy/structos__timer__def.html) implementation based on either the [low power ticker](../mbed-os-api-doxy/group__hal__lp__ticker.html) or [microsecond ticker](../mbed-os-api-doxy/group__hal__us__ticker.html) if "tickless-from-us-ticker" target configuration is set. This change is necessary to avoid drift connected with using two different timers to measure time. Users must not change the ticker interrupt handler when tickless mode is in use.
+
+### Enable tickless mode support in bare metal profile 
+
+Tickless mode is enabled by default for bare metal profile without adding `MBED_TICKLESS` target configuration. As mentioned earlier target prefer to use microsecond ticker for tickless mode or does not have low power ticker then add the below target configuration in `targets.json`.
+
+```
+"overrides": {
+    "tickless-from-us-ticker": true
+}
+```
+
+### Testing
+
+There are no dedicated tests validating tickless mode. Running all Mbed OS tests suits, with particular focus on HAL sleep and HAL low power ticker tests, provides sufficient coverage.
```

Comment:
This paragraph might need rewriting. Currently we do not have dedicated tests but we could (for example we should ensure that there is no clock drift. We could also ensure that higher power savings are achieved). Are we sure that the current HAL tests provide enough coverage?

Reply:
What is missing at the end is some notes on current issues/challenges with tickless. @gpsimenos can you track why your changes to enable tickless were reverted?